### ISSUE TITLE
Thank you!

### ISSUE THREAD

**Issue by overvalidated** [apple/ml-cross-entropy#7]
I had this idea almost year ago, trying to train mT5 and realizing that LM head takes 30gb of RAM. My company seeks mostly profit and it was impossible to continue to work on it (and also my implementation was slow, I didn't use gradient filtering, thinking about online softmax, not that it is useless to calculate most of it). But now I can use your implementation and to be grateful for the immense work you've put to accomplish this.

It actually gives us opportunity to optimize things further and create a new pretrained model, pushing our model quality further.

Thank you!

**Comment by erikwijmans** [comment]
Your welcome! Glad you like it and find it useful.
